vllm/datadog_checks/vllm/check.py

```diff
@@ -0,0 +1,94 @@
+        self._health_endpoint = DEFAULT_HEALTH_ENDPOINT
+
+        # Get the API server port if specified, otherwise use the default 8000.
+        self.server_port = str(self.instance.get('server_port', 8000))
+
+        self.collect_server_info = self.instance.get('collect_server_info', True)
+
+        # Get the base url from the openmetrics endpoint and construct the server info API endpoint.
+        if self.collect_server_info and self.openmetrics_endpoint:
+            parts = urlparse(self.openmetrics_endpoint)
+            # Delete the /metrics from the url
+            self.base_url = parts._replace(path="")
+            # Replace the openmetrics port by the server port
+            self.server_info_api = self.base_url._replace(netloc=parts.hostname + ':' + self.server_port).geturl()
+        else:
+            self.log.debug("Collecting server info through API is disabled.")
+            return
+
+    def get_default_config(self):
+        return {
+            'metrics': [METRIC_MAP],
+            "rename_labels": RENAME_LABELS_MAP,
+        }
+
+    @AgentCheck.metadata_entrypoint
+    def _submit_version_metadata(self):
+
+        endpoint = self.instance.get("openmetrics_endpoint").replace("/metrics", DEFAULT_VERSION_ENDPOINT)
+        response = self.http.get(endpoint)
+
+        if response.ok:
+            data = response.json()
+            version = data.get("version", "")
+            version_split = version.split(".")
+            if len(version_split) >= 3:
+                major = version_split[0]
+                minor = version_split[1]
+                patch = version_split[2]
+
+                version_raw = f'{major}.{minor}.{patch}'
+
+                version_parts = {
+                    'major': major,
+                    'minor': minor,
+                    'patch': patch,
+                }
+                self.set_metadata('version', version_raw, scheme='semver', part_map=version_parts)
+            else:
+                self.log.debug("Invalid Weaviate version format: %s", version)
+        else:
+            self.log.debug("Could not retrieve version metadata from host.")
+
+    def _check_server_health(self):
+        endpoint = urljoin(self.server_info_api, self._health_endpoint)
+        response = self.http.get(endpoint)
+
+        # Any 4xx or 5xx response from the API endpoint (/v2/health/ready) means the server is not ready
+        if 400 <= response.status_code and response.status_code < 600:
+            self.service_check('health.status', AgentCheck.CRITICAL, self.tags)
+        if response.status_code == 200:
+            self.service_check('health.status', AgentCheck.OK, self.tags)
+        else:
+            self.service_check('health.status', AgentCheck.UNKNOWN, self.tags)
+
+    def check(self, instance):
+        super().check(instance)
+        if self.collect_server_info:
+            self._submit_version_metadata()
+            self._check_server_health()
```

Comment:
What if we just try to submit the version metadata no matter what?

It would make config and implementation simpler and send more information by default. If there are real user requests to make it optional, then we add the complexity.

Reply:
The reason I opted not to is that it's unclear whether or not the API and `/metrics` always share the same port or can be configured. If so, it makes sense to force user to enable it themselves rather than erroring. We've taken both approaches in other checks/integrations so if you feel strongly about it, I can make those changes